### PR TITLE
Support regex selection in SHOW TAG VALUES for the key

### LINKED ISSUE #4532: [feature request] SHOW TAG VALUES should accept regex
Right now the syntax is `SHOW TAG VALUES WITH KEY IN (<tag_key1>, <tag_key2>,...)` which is not congruent with our other queries and doesn't support regular expressions.

How about `SHOW TAG VALUES WHERE KEY [=,=~,!=,!~] <tag_key_regex>`

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #4532.

### CODE REVIEW

File: influxql/ast.go [hunk 1]
```diff
@@ -3247,6 +3249,25 @@ func isFalseLiteral(expr Expr) bool {
 	return false
 }
 
+// SetLiteral represents a set literal.
+type SetLiteral struct {
+	Vals []string
+}
```
Comment (benbjohnson) [inline influxql/ast.go]: Should this be a `ListLiteral` if we're not ensuring uniqueness?

**Comment by benbjohnson** [comment]
Minor naming comment otherwise lgtm

File: influxql/ast.go (hunk 1 above)
Comment (jsternberg) [inline influxql/ast.go]: I wanted this to eventually be an actual set to implement something else, but I think it's reasonable to rename this to `ListLiteral` until that change is actually done.

**Comment by corylanou** [comment]
+1

**Pull request merged**